TutorialApplication/OrganizationInfoViewController.swift

```diff
@@ -14,6 +14,87 @@ import UIKit
+        
+        for jsonObjects in organizations as! [AnyObject]
+            {
+                let organizationName = jsonObjects["name"] as! String
+                
+                let organizationEmployees = jsonObjects["employees"] as! [[String : AnyObject]]
+                
+                print("org name = \(organizationName)  employess = \(organizationEmployees)")
+                
+                newOrganization = NSEntityDescription.insertNewObject(forEntityName: "FFOrganization", into: DatabaseController.sharedInstance().context) as! FFOrganization
+                
+                newOrganization.name = organizationName
+                
+                for item in organizationEmployees
+                {
+                    let dateFormatter = DateFormatter()
+                    dateFormatter.dateFormat = "yyyy-MM-dd"
+                    
+                    newEmployee = NSEntityDescription.insertNewObject(forEntityName: "FFEmployee", into: DatabaseController.sharedInstance().context) as! FFEmployee
+
+                    newEmployee.firstName = (item["first_name"] as! String)
+                    newEmployee.lastName = (item["last_name"] as! String)
+                    newEmployee.order = item["order"] as! Int16
+                    
+                    if item["salary"] != nil && !(item["salary"] is NSNull)
+                    {
+                        newEmployee.salary = item["salary"] as! Int32
+                    }
+                    
+                    newEmployee.isActive = item["isActive"] as! Bool
+                    newEmployee.dateOfBirth = dateFormatter.date(from:"1986-04-13")
+                    newOrganization.addEmployeesObject(newEmployee)
+                }
+                 DatabaseController.saveContext()
+            }
+        }
+            
+            
+        let actionSheetController = UIAlertController(title: "Please select organization", message: "", preferredStyle: .actionSheet)
+            
+        let cancelActionButton = UIAlertAction(title: "Cancel", style: .cancel)
+        { action -> Void in
+            print("Cancel")
+        }
+        actionSheetController.addAction(cancelActionButton)
+            
+            for organizations in 0...Int((countOfOrganizations?.count)!)-1
+            {
+                let organizationsList = UIAlertAction(title: (countOfOrganizations?[organizations] as! FFOrganization).name, style: .default)
+                { action -> Void in
+                    
+                    let pickedOrganization = countOfOrganizations?[organizations] as! FFOrganization
+                    
+                    NotificationCenter.default.post(name: Notification.Name(rawValue: OrganizationInfoViewController.kOrganizationHasChanged), object: self, userInfo: ["organization" : pickedOrganization])
+                    
+                    self.navigationController!.popViewController(animated: true)
```

Comment:
Calling self inside a block or closure may result in retain cycle. Read about it and fix it.